plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -272,30 +287,58 @@ def alter_retention_policy(module, client, retention_policy):
     module.exit_json(changed=changed)
 
 
+def drop_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+
+    changed = False
+
+    if not module.check_mode:
+        try:
+            client.drop_retention_policy(policy_name, database_name)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+        changed = True
+    module.exit_json(changed=changed)
+
+
 def main():
     argument_spec = InfluxDb.influxdb_argument_spec()
     argument_spec.update(
+        state=dict(default='present', type='str', choices=['present', 'absent']),
         database_name=dict(required=True, type='str'),
         policy_name=dict(required=True, type='str'),
-        duration=dict(required=True, type='str'),
-        replication=dict(required=True, type='int'),
+        duration=dict(required=False, type='str'),
+        replication=dict(required=False, type='int'),
```

Comment:
```suggestion
        duration=dict(type='str'),
        replication=dict(type='int'),
```

Reply:
There is a third argument, `shard_group_duration`, not modified in this PR, that also has `required=False`. I added it in both `duration` and `replication` to "standardize" format. Should I remove it from that argument too?

Reply:
Yes, please modify that one as well. Thanks!